src/lib.rs

```diff
@@ -79,18 +79,38 @@ use instant::Instant;
 #[cfg(not(target_arch = "wasm32"))]
 use std::time::Instant;
 
+pub trait Seed {
+    fn with_seed(seed: u64) -> Rng<Self> where Self: Sized;
+    fn get(&self) -> u64;
+    fn set(&self, new_seed: u64);
+}
+
+// Seed abstraction, so users can specify their own
+#[derive(Debug, PartialEq, Eq)]
+pub struct CellSeed(Cell<u64>);
```

Comment:
I don't think the new-type struct abstraction is necessary, it imo should be omitted. It might be a good idea to replace it with a type alias.
```rust
pub type CellSeed = Cell<u64>;
```